columnphysics/icepack_atmo.F90

```diff
@@ -271,37 +267,43 @@ psixh     ! stability function at zlvl   (heat and water)
       ! iterate to converge on Z/L, ustar, tstar and qstar
       !------------------------------------------------------------
 
+      ustar_prev = c2 * ustar
+
       do k = 1, natmiter
 
-         ! compute stability & evaluate all stability functions
-         hol = vonkar * gravit * zlvl &
-                 * (tstar/thva &
-                 + qstar/(c1/zvir+Qa)) &
-                 / ustar**2
-         hol    = sign( min(abs(hol),c10), hol )
-         stable = p5 + sign(p5 , hol)
-         xqq    = max(sqrt(abs(c1 - c16*hol)) , c1)
-         xqq    = sqrt(xqq)
-
-         ! Jordan et al 1999
-         psimhs = -(0.7_dbl_kind*hol &
-                + 0.75_dbl_kind*(hol-14.3_dbl_kind) &
-                * exp(-0.35_dbl_kind*hol) + 10.7_dbl_kind)
-         psimh  = psimhs*stable &
-                + (c1 - stable)*psimhu(xqq)
-         psixh  = psimhs*stable &
-                + (c1 - stable)*psixhu(xqq)
-
-         ! shift all coeffs to measurement height and stability
-         rd = rdn / (c1+rdn/vonkar*(alz-psimh))
-         rh = rhn / (c1+rhn/vonkar*(alz-psixh))
-         re = ren / (c1+ren/vonkar*(alz-psixh))
+         if (.not. (natmiter_conv > c0 .and. &
+                 abs(ustar - ustar_prev)/ustar <= natmiter_conv)) then
```

Comment:
A couple questions re lines 270-275 in icepack_atmo.F90:
The value of ustar_prev being compared with seems a little arbitrary.  Why 2*ustar?
It looks like natmiter_conv does not change during the loop, so it would be more efficient to evaluate that part of the conditional outside of the loop.  Does that make sense?

Reply:
This is just to initialize the convergence loop and make sure it does it at least once.

Reply:
Is it ever possible for ustar=0 on the first iteration, or thereafter?
Is ustar ever negative? 
For efficiency, could we remove the check on natmiter_conv and rewrite the lines 272-275 as
`      do k = 1, natmiter`     
`             if (abs((ustar - ustar_prev)/ustar) > natmiter_conv)) then`
?
Also, now that I'm looking at this, I don't like the 'n' at the beginning of natmiter_conv.  That denotes a number (i.e. integer, the number of iterations, in this case) but the _conv variable is a tolerance rather than an integer.  It's only a little confusing to those of us (maybe only me) who knows the convention for variable naming in CICE, which was inherited from POP...

Reply:
Ustar (friction velocity) in the atmosphere could in theory be zero but it can never be negative.

Reply:
ustar as it is currently set cannot be zero.  It is dependent on the minimum allowable wind speed, umin, which is greater than zero in the initialization of atmo_boundary_layer:

 185       if (highfreq) then
 186        umin  = p5 ! minumum allowable wind-ice speed difference of      0.5 m/s
 187       else
 188        umin  = c1 ! minumum allowable wind speed of 1m/s
 189       endif

This limit is to ensure that sensible and latent heat flux are not zero when the true wind magnitude are zero - if umin is ever set to zero, this breaks the science as well as the code.  Wind stress magnitude can still be zero, but that's performed separately.

Reply:
> Also, now that I'm looking at this, I don't like the 'n' at the beginning of natmiter_conv. That denotes a number (i.e. integer, the number of iterations, in this case) but the _conv variable is a tolerance rather than an integer. 

I agree with Elizabeth on this.

Reply:
Thanks for taking a look at this, @proteanplanet.
@dabail10 it looks like there are 2 things remaining to address:  removing n from natmiter_conv, and changing the conditional as noted above to (potentially) make the loop more efficient.  For the latter, let's make sure that my logic is correct...